.github/workflows/build.yaml

```diff
@@ -0,0 +1,43 @@
+name: "CI-basic"
+on:
+  pull_request:
+  push:
+    branches:
+      - master
+env:
+  GITHUB_PACKAGE_REGISTRY_TOKEN: ${{ secrets.GH_PACKAGES }}
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    container: docker://mayadata/ms-buildenv:nix
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+      - run: cargo build --all
+  Clippy:
+    runs-on: ubuntu-latest
+    container: docker://mayadata/ms-buildenv:nix
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions-rs/toolchain@v1
+        with:
+          toolchain: nightly
+          components: clippy
+      - uses: actions-rs/clippy-check@v1.0.5
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          args: --all --all-targets -- -D warnings
+  nix-build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: cachix/install-nix-action@v7
+      - run: nix-build -A mayastorImage
```

Comment:
Aren't we building this on line 10-18 above?

Reply:
This builds a docker image, the lines above build the code in 'debug' mode. So they are strong related and we can use this approach above as well to create "debug images" but - we we can not run the checks this way. (unless we hack the actions themselves).